### ISSUE TITLE
optimize 'edit history dataset' request flow

### ISSUE THREAD

**Issue by martenson** [galaxyproject/galaxy#4795]
Currently the get_edit endpoint loads ~100kB of data for each request because all dbkeys and all file_types are included in the response as valid options.

We have dedicated API endpoints for these data that provide much better caching options. The whole (analysis) app should also store these objects in memory for sharing amongst all use cases.

related: #4779

**Comment by guerler** [comment]
Thanks for opening an issue. These are the same lists used for the upload view.

**Comment by martenson** [comment]
@guerler what do you mean by 'the same lists'? That they have same contents? 

These lists are also used in data libraries via e.g. https://docs.galaxyproject.org/en/master/api/api.html#module-galaxy.webapps.galaxy.api.genomes

**Comment by guerler** [comment]
I mean the build and file type/extension selector in the upload view. They should be equivalent or at least very similar.

**Comment by martenson** [comment]
@guerler agreed, there should be one instance of each of these lists across the app
